src/main/java/org/broadinstitute/hellbender/tools/spark/ParallelCopyGCSDirectoryIntoHDFSSpark.java

```diff
@@ -0,0 +1,200 @@
+
+            final Iterator<Integer> iterator = chunkMap.keySet().iterator();
+            while (iterator.hasNext()) {
+                final Integer next = iterator.next();
+                final String chunkPath = chunkMap.get(next);
+                chunkPaths[next] = new org.apache.hadoop.fs.Path(chunkPath);
+            }
+
+            fs.concat(outFilePath, chunkPaths);
+        }
+    }
+
+    private List<Tuple2<String, Integer>> setupChunks(final long chunkSize, final List<Path> gcsNIOPaths) throws IOException {
+        List<Tuple2<String, Integer>> chunkList = new ArrayList<>();
+        for (Path path : gcsNIOPaths) {
+
+            if (Files.isDirectory(path)) {
+                logger.info("skipping directory " + path);
+                continue;
+            }
+
+            final long fileSize = Files.size(path);
+            final long chunks = fileSize / chunkSize + (fileSize % chunkSize == 0 ? 0 : 1);
+            logger.info("processing path " + path + ", size = " + fileSize + ", chunks = " + chunks);
+
+            for (int i = 0; i < chunks; i++) {
+                chunkList.add(new Tuple2<>(path.toUri().toString(), i));
+            }
+        }
+        return chunkList;
+    }
+
+    private List<Path> getGCSFilePathsToCopy(final String inputGCSPathFinal) throws IOException {
+        final List<Path> gcsNIOPaths;
+        final Path inputGCSNIOPath = IOUtils.getPath(inputGCSPathFinal);
+        if (Files.isDirectory(inputGCSNIOPath)) {
+            logger.info("transferring input directory: " + inputGCSPathFinal);
+            gcsNIOPaths = Files.list(inputGCSNIOPath).collect(Collectors.toList());
+        } else {
+            logger.info("transferring single file: " + inputGCSNIOPath);
+            gcsNIOPaths = Collections.singletonList(inputGCSNIOPath);
+        }
+        return gcsNIOPaths;
+    }
+
+    static long getChunkSize(final FileSystem fs) {
+        return Long.parseLong(fs.getConf().get("dfs.blocksize"));
+    }
+
+    private static final Tuple2<Integer, String> readChunkToHdfs(final String inputGCSPathFinal, final long chunkSize, final Integer chunkNum, final String outputDirectory) {
+        final Path gcsPath = IOUtils.getPath(inputGCSPathFinal);
+        final String basename = gcsPath.getName(gcsPath.getNameCount() - 1).toString();
+        org.apache.hadoop.fs.Path outputPath = new org.apache.hadoop.fs.Path(outputDirectory);
+        final String chunkPath = outputPath + "/" + basename + ".chunk." + chunkNum;
+
+        try (SeekableByteChannel channel = Files.newByteChannel(gcsPath);
+             final OutputStream outputStream = new BufferedOutputStream(BucketUtils.createNonGCSFile(chunkPath))){
+
+            final long start = chunkSize * (long) chunkNum;
+            channel.position(start);
+            ByteBuffer byteBuffer = ByteBuffer.allocateDirect((int) Math.min(67108864, chunkSize));
```

Comment:
Document where this magic value `67108864` comes from, and extract as a named constant.